tasks/init.rb

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative "../../ruby_task_helper/files/task_helper.rb"
+
+require 'net/http'
+require 'openssl'
+require 'json'
+
+class HTTPRequest < TaskHelper
+  def task(**opts)
+    method  = opts[:method].capitalize.to_sym
+    uri     = URI.parse(opts[:url])
+    headers = make_headers(opts[:headers], opts[:body_format], opts[:user_agent])
+    body    = make_body(opts[:body], opts[:body_format])
+    options = opts.slice(:cacert, :cert, :key)
+
+    response = request(method, uri, headers, body, options)
+    handle_response(response, opts[:success_codes])
+  rescue TaskHelper::Error => e
+    { _error: e.to_h }
+  end
+
+  private
+
+  # Builds the headers for the request, setting the Content-Type according to
+  # the value of :body_format. If Content-Type is specified in :headers, that
+  # value will take priority.
+  #
+  def make_headers(headers, body_format, user_agent)
+    base = case body_format
+           when 'form-urlencoded'
+            { 'Content-Type' => 'application/x-www-form-urlencoded' }
+           when 'json'
+            { 'Content-Type' => 'application/json' }
+           when 'raw'
+            { 'Content-Type' => 'text/plain' }
+           end
+
+    base['User-Agent'] = user_agent
+
+    base.merge!(headers || {})
+  end
+
+  # Formats the body for the request. If the Content-Type is application/json,
+  # this will convert the body to JSON. Otherwise, the body will be stringified.
+  #
+  def make_body(body, body_format)
```

Comment:
`body_format` will set a default `Content-Type` header, which can be overridden by setting the same header under `headers`. I'm not sure if it's more appropriate to check the `body_format` or `Content-Type` header when determining how to format the body...